packages/client/test/rpc/eth/getProof.spec.ts

```diff
@@ -0,0 +1,137 @@
+            }
+        }
+    */
+  const data =
+    '0x6080604052348015600f57600080fd5b5060bc8061001e6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063975057e714602d575b600080fd5b60336035565b005b6104d260008190555061162e600160003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000208190555056fea2646970667358221220b16fe0abdbdcae31fa05c5717ebc442024b20fb637907d1a05547ea2d8ec8e5964736f6c63430007060033'
+
+  // construct tx
+  const tx = Transaction.fromTxData({ gasLimit: 2000000, data }, { freeze: false })
+  tx.getSenderAddress = () => {
+    return address
+  }
+
+  // deploy contract
+  const { createdAddress } = await vm.runTx({ tx })
+
+  // call store() method
+  const funcHash = '975057e7' // store()
+  const storeTxData = {
+    to: createdAddress!.toString(),
+    from: address.toString(),
+    data: `0x${funcHash}`,
+    gasLimit: `0x${new BN(530000).toString(16)}`,
+  }
+  const storeTx = Transaction.fromTxData(storeTxData, { freeze: false })
+  storeTx.getSenderAddress = () => {
+    return address
+  }
+  await vm.runTx({
+    tx: storeTx,
+    skipNonce: true,
+    skipBalance: true,
+    skipBlockGasLimitValidation: true,
+  })
+
+  return { server, createdAddress: createdAddress! }
+}
+
+tape(`${method}: call with valid arguments (retrieve pos0)`, async (t) => {
+  const { server, createdAddress } = await setup()
+
+  // verify storage of pos0 is accurate
+  const req = params(method, [createdAddress!.toString(), ['0x0'], 'latest'])
+  const expectRes = (res: any) => {
+    const msg = 'should return the correct storage value'
+    t.deepEqual(res.body.result, expectedProof, msg)
+  }
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(`${method}: call with unsupported block argument`, async (t) => {
+  const { server, createdAddress } = await setup()
+
+  const req = params(method, [createdAddress!.toString(), ['0x0'], 'pending'])
+  const expectRes = checkError(t, INVALID_PARAMS, 'Currently only "latest" block supported')
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(`${method}: call with unsupported storage slots argument`, async (t) => {
+  const { server, createdAddress } = await setup()
+
+  const req = params(method, [createdAddress!.toString(), '0x0', 'latest'])
+  const expectRes = checkError(t, INVALID_PARAMS, 'invalid argument 1: argument must be an array')
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(
+  `${method}: call with unsupported storage slots argument: array without hex strings`,
+  async (t) => {
+    const { server, createdAddress } = await setup()
+
+    const req = params(method, [createdAddress!.toString(), ['0'], 'latest'])
+    const expectRes = checkError(
+      t,
+      INVALID_PARAMS,
+      'invalid argument 1: invalid argument 0: hex string without 0x prefix'
+    )
+    await baseRequest(t, server, req, 200, expectRes)
+  }
+)
```

Comment:
Perhaps we could also add a test with the latest block number (instead of the string 'latest') as third param?

Reply:
i think this is okay, if you look at my new code in #1598 it normalizes the block option lookup (and is tested within `getBalance.spec.ts` within that PR) so it should have adequate coverage.